src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml

```diff
@@ -106,15 +110,96 @@ limitations under the License.
     <item fieldId="2" name="CoolSetpoint" type="int16s" isNullable="true"/>
   </struct>
 
+  <bitmap name="PresetTypeFeaturesBitmap" type="bitmap8">
+    <cluster code="0x0201"/>
+    <field name="Automatic" mask="0x01"/>
+    <field name="SupportsNames" mask="0x02"/>
+  </bitmap>
+
+  <enum name="PresetScenarioEnum" type="enum8">
+    <cluster code="0x0201"/>
+    <item name="Unspecified" value="0x00"/>
+    <item name="Occupied" value="0x01"/>
+    <item name="Unoccupied" value="0x02"/>
+    <item name="Sleep" value="0x03"/>
+    <item name="Wake" value="0x04"/>
+    <item name="Vacation" value="0x05"/>
+    <item name="UserDefined" value="0x06"/>
+  </enum>
+
+  <struct name="PresetTypeStruct">
+    <cluster code="0x201"/>
+    <item fieldId="0" name="PresetScenario" type="PresetScenarioEnum"/>
+    <item fieldId="1" name="NumberOfPresets" type="int8u"/>
+    <item fieldId="2" name="PresetTypeFeatures" type="PresetTypeFeaturesBitmap"/>
+  </struct>
+
+  <struct name="PresetStruct">
+    <cluster code="0x201"/>
+    <item fieldId="0" name="PresetHandle" type="octet_string" length="16"/>
+    <item fieldId="1" name="PresetScenario" type="PresetScenarioEnum"/>
+    <item fieldId="2" name="Name" type="char_string" length="64" isNullable="true" default="null"/>
+    <item fieldId="3" name="CoolingSetpoint" type="int16s" default="2600"/>
+    <item fieldId="4" name="HeatingSetpoint" type="int16s" default="2000"/>
+    <item fieldId="5" name="BuiltIn" type="boolean" default="false"/>
+  </struct>
+
+  <struct name="QueuedPresetStruct">
+    <cluster code="0x201"/>
+    <item fieldId="0" name="PresetHandle" type="octet_string" length="16"/>
+    <item fieldId="1" name="TransitionTimestamp" type="epoch_s"/>
+  </struct>
+
+  <bitmap name="ScheduleTypeFeaturesBitmap" type="bitmap8">
+    <cluster code="0x0201"/>
+    <field name="SupportsPresets" mask="0x01"/>
+    <field name="SupportsSetpoints" mask="0x02"/>
+    <field name="SupportsNames" mask="0x03"/>
+    <field name="SupportsOff" mask="0x04"/>
+  </bitmap>
+
+  <struct name="ScheduleTypeStruct">
+    <cluster code="0x201"/>
+    <item fieldId="0" name="SystemMode" type="ThermostatSystemMode"/>
+    <item fieldId="1" name="NumberOfSchedules" type="int8u" default="0"/>
+    <item fieldId="2" name="ScheduleTypeFeatures" type="ScheduleTypeFeaturesBitmap"/>
+  </struct>
+
+  <struct name="ScheduleTransitionStruct">
```

Comment:
This has optional fields....

I am stopping here.  Either this is not implementing the spec I am reading, or it matches the spec so badly that there are literally errors in almost every single thing added....